### PR TITLE
init method added to CaseInsensitiveDict

### PR DESCRIPTION
**Pull request by rozuur**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the related issue number (starting with `#`)**
#1231 

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

**Comment by webknjaz** [comment]
This pull request introduces 1 alert - [view on lgtm.com](https://lgtm.com/projects/g/cherrypy/cherrypy/rev/pr-64a4487de51daeae4f2f9af6276176316bd67d67)

**new alerts:**

* 1 for Missing call to __init__ during object initialization

---

*Comment posted by [lgtm.com](https://lgtm.com)*

File: cherrypy/lib/cptools.py [hunk 1]
```diff
@@ -586,6 +586,7 @@ def accept(media=None, debug=False):
 class MonitoredHeaderMap(_httputil.HeaderMap):
 
     def __init__(self):
+        super(MonitoredHeaderMap, self).__init__()
```
Comment (webknjaz) [inline cherrypy/lib/cptools.py:589]: Superclass's `__init__(*args, **kwargs)` has different signature. Please write test for this.

**Review by webknjaz — CHANGES_REQUESTED** [review]
Looks good, but please add tests for these changes.

File: cherrypy/lib/cptools.py (hunk 1 above)
Comment (rozuur) [reply cherrypy/lib/cptools.py:589]: Have taken the liberty of writing test case at cherrypy/test/test_misc_tools.py, please review

File: cherrypy/lib/httputil.py [hunk 2]
```diff
@@ -380,6 +380,10 @@ class CaseInsensitiveDict(dict):
 
     Each key is changed on entry to str(key).title().
     """
+    def __init__(self, *args, **kwargs):
```
Comment (webknjaz) [inline cherrypy/lib/httputil.py:383]: Plz test this in `test_httputil` test module

File: cherrypy/lib/cptools.py (hunk 1 above)
Comment (webknjaz) [reply cherrypy/lib/cptools.py:589]: still, you don't use args/kwargs, add them

File: cherrypy/test/test_misc_tools.py [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import os
+import unittest
```
Comment (webknjaz) [inline cherrypy/test/test_misc_tools.py:2]: you don't need this

Comment (webknjaz) [reply cherrypy/test/test_misc_tools.py:2]: use `pytest` instead

File: cherrypy/test/test_misc_tools.py [hunk 4]
```diff
@@ -213,3 +214,9 @@ def testAutoVary(self):
             'Accept, Accept-Charset, Accept-Encoding, '
             'Host, If-Modified-Since, Range'
         )
+
+
+class MonitoredHeaderMapTest(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (webknjaz) [reply cherrypy/test/test_misc_tools.py:219]: inherit from object and start class name with `Test` as required by `pytest`

File: cherrypy/test/test_misc_tools.py [hunk 5]
```diff
@@ -213,3 +214,9 @@ def testAutoVary(self):
...
+    def test_creation(self):
```
[the remark this answers is not in the document]
Comment (webknjaz) [reply cherrypy/test/test_misc_tools.py:220]: plz compare different ways of creation here (including with/without arguments)

[the remark this answers is not in the document]
Comment (webknjaz) [reply cherrypy/test/test_misc_tools.py:220]: + test header access tracking

**Comment by jaraco** [comment]
Thank you @rozuur for this PR, and while your change is a good one, I've decided I'd prefer to move forward with the approach in #1654 instead.

**Pull request closed without merging**